extensions/roulette/unmute/cog.py

```diff
@@ -0,0 +1,88 @@
+import logging
+
+from . import action
+from ..config import config
+from ..config.redis_config import redis_client
+
+from datetime import datetime
+from discord import Role
+from discord.ext import tasks
+from discord.ext.commands import Bot, Cog
+from typing import Set
+
+
+class Unmute(Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.logger = logging.getLogger("roulette.unmute")
+        self.unmute_loop.change_interval(minutes=config.unmute_rate())
+        self.unmute_loop.start()
+        self.logger.info("Loaded Unmute cog")
+
+    async def cog_command_error(self, ctx, error: Exception) -> None:
+        self.logger.warning(str(error))
+
+    @tasks.loop(minutes=1)
+    async def unmute_loop(self):
+        await self._unmute(await self._get_guild_ids())
+
+    async def _get_guild_ids(self) -> Set[int]:
+        cursor = 0
+        pattern = 'roulette_timeout_live*'
+
+        guild_keys = set()
+        guild_ids = set()
+
+        while True:
+            cursor, partial_keys = redis_client.scan(cursor=cursor, match=pattern)
+            guild_keys.update(partial_keys)
+
+            if cursor == 0:
+                break
+
+        for key in guild_keys:
+            guild_ids.add(int(key[22:]))
```

Comment:
nit: Don't use [22:] here; it's not particularly readable. Prefer to reference the pattern again if you can.